app/models/ogi_pilot_session.rb

```diff
@@ -1,6 +1,128 @@
 class OgiPilotSession < ApplicationRecord
+
+  has_secure_password
   validates :end_point, presence: true
   validates :channel, presence: true
   validates :topic, presence: true
   validates :message_type, presence: true
+  validates :user_name, presence: true
+
+  attr_accessor :validation_messages
+
+  def provider_session_exists
+    return self.provider_session_id.present?
+  end
+
+  def consumer_session_exists
+    return self.consumer_session_id.present?
+  end
```

Comment:
From a style perspective I would call these either `provider_session?` or `provider_session_exists?` or even `provider_session_present?` (for consistency with the other terminology around presence).
The question mark convention is used to indicate that the method provides a boolean.
Also, the style conventions of ruby avoids using 'self' unless necessary. So you usually only need to use `self` for assignment to differentiate it from a local variable assignment.